components/breakpoints.ts

```diff
@@ -1,6 +1,49 @@
-export const maxWidth = (breakpoint) => `max-width: ${breakpoint}px`;
-export const minWidth = (breakpoint) => `min-width: ${breakpoint + 1}px`;
+// <: extra small devices - portrait phones
+const BREAKPOINT_1 = 576;
+// > <: small devices - landscape phones
+const BREAKPOINT_2 = 768;
+// > <: medium devices - tablets
+const BREAKPOINT_3 = 992;
+// > <: large devices - desktops
+const BREAKPOINT_4 = 1200;
+// > <: extra large devices - large desktops, etc.
+const BREAKPOINT_5 = 1600;
+// >: extra extra large devices - Big McLargeHuge devices
 
-export const BREAKPOINT_DESKTOP = 1024;
-export const BREAKPOINT_IPAD = 768;
-export const BREAKPOINT_MOBILE = 576;
+function breakUpPx(breakpoint) {
+  switch (breakpoint) {
+    case "sm":
+      return BREAKPOINT_1;
+    case "md":
+      return BREAKPOINT_2;
+    case "lg":
+      return BREAKPOINT_3;
+    case "xl":
+      return BREAKPOINT_4;
+    case "xxl":
+      return BREAKPOINT_5;
+    default:
+      return breakpoint;
+  }
+}
+
+function breakDownPx(breakpoint) {
+  switch (breakpoint) {
+    case "xs":
+      return BREAKPOINT_1;
+    case "sm":
+      return BREAKPOINT_2;
+    case "md":
+      return BREAKPOINT_3;
+    case "lg":
+      return BREAKPOINT_4;
+    case "xl":
+      return BREAKPOINT_5;
+    default:
+      return breakpoint;
+  }
+}
+
+export const breakUp = (breakpoint) => `min-width: ${breakUpPx(breakpoint)}px`;
```

Comment:
Nice! I like how we don't have to import the breakpoint anymore and can just pass a string.

We might also want to add a type to `breakpoint` to ensure we pass a valid breakpoint to it:

```typescript
type Breakpoint = "xs" | "sm" | "md" | "lg" | "xl" | "xxl"

export const breakUp = (breakpoint: Breakpoint) => `min-width: ${breakUpPx(breakpoint)}px`;
```

Reply:
I just pushed up a commit that adds types. It does remove the ability to pass in a custom width, which I was sort of intending to support if possible, but I think it's ok without.